src/Serialization/Rings.jl

```diff
@@ -0,0 +1,111 @@
+################################################################################
+#  non simpleton base rings
+
+function load_internal(s::DeserializerState, t::Type{Nemo.NmodRing}, dict::Dict)
+    modulus = load_type_dispatch(s, UInt64, dict[:modulus])
+    
+    return t(modulus)
+end
+
+function save_internal(s::SerializerState, R::Nemo.NmodRing)
+    return Dict(
+        :modulus => save_type_dispatch(s, modulus(R))
+    )
+end
+
+################################################################################
+#  Polynomial Rings
+function save_internal(s::SerializerState, R::Union{MPolyRing, PolyRing})
+    return Dict(
+        :symbols => save_type_dispatch(s, symbols(R)),
+        :base_ring => save_type_dispatch(s, base_ring(R)),
+    )
+end
+
+function load_internal(s::DeserializerState,
+                       T::Type{<: Union{MPolyRing, PolyRing}},
+                       dict::Dict)
+    base_ring = load_type_dispatch(s, dict[:base_ring], check_namespace=false)
+    symbols = load_type_dispatch(s, Vector{Symbol}, dict[:symbols]) 
+
+    if T isa PolyRing
```

Comment:
Sorry, I made a bad suggestion: of course `T` is a type, not a ring, so I think you actually need this:
```suggestion
    if T <: PolyRing
```